src/test_quadratic.cpp

```diff
@@ -0,0 +1,119 @@
+#include "solve_quadratic.h"
+
+void TestCompareDouble () {
+    printf ("\n-----------------------------------------------------------------------------\n");
+    printf (  "-----------------Now-start-Unit-Testing-of-compare-of-doubles----------------\n");
+    printf (  "-----------------------------------------------------------------------------\n");
+
+    double tests[10][3] = { {         1e9,           0,  1},
```

Comment:
Why 10 and 3 (magic numbers)? Also, I'd make a structure (something like CompareDoubleTestCase)